packages/shell/src/cli-run.ts

```diff
@@ -117,3 +147,15 @@ async function setupHistory(replServer: repl.REPLServer) {
     // Ignore. It's ok not to have a history file
   }
 }
+
+function parseDatabaseURL(databaseURL?: string) {
+  databaseURL = databaseURL || getDatabaseURL() || '';
+  const [protocol, , host, , database] = databaseURL.split('/');
+  const [workspace] = (host || '').split('.');
+  return {
+    protocol,
+    host,
+    database,
+    workspace
+  };
+}
```

Comment:
This is from the CLI, I think more and more a util package would be appreciated 😄

Reply:
Yep. Or we can put it in the `xata/client` one.